src/Popper.js

```diff
@@ -68,22 +68,22 @@ export class InnerPopper extends React.Component<PopperProps, PopperState> {
     placement: undefined,
   };
 
-  popperInstance: ?Instance;
+  popperInstance: Instance | null;
 
-  popperNode: ?HTMLElement = null;
-  arrowNode: ?HTMLElement = null;
+  popperNode: HTMLElement | null = null;
+  arrowNode: HTMLElement | null = null;
 
-  setPopperNode = (popperNode: ?HTMLElement) => {
-    if (this.popperNode === popperNode) return;
+  setPopperNode = (popperNode: HTMLElement | null) => {
+    if (!popperNode || this.popperNode === popperNode) return;
 
     safeInvoke(this.props.innerRef, popperNode);
     this.popperNode = popperNode;
 
     this.updatePopperInstance();
   };
 
-  setArrowNode = (arrowNode: ?HTMLElement) => {
-    if (this.arrowNode === arrowNode) return;
+  setArrowNode = (arrowNode: HTMLElement | null) => {
+    if (!arrowNode || this.arrowNode === arrowNode) return;
     this.arrowNode = arrowNode;
 
     if (!this.popperInstance) this.updatePopperInstance();
```

Comment:
Don't quite understand way we have this code, in what situation we will set the arrow and we don't have the `popperInstance` ?  IMHO would simplify it and just use 

```js
  setArrowNode = (arrowNode: HTMLElement | null) => {
    this.arrowNode = arrowNode;
  };
```